engine/gameobject/src/gameobject/test/anim/test_gameobject_anim.cpp

```diff
@@ -221,19 +235,28 @@ TEST_F(AnimTest, Cancel)
     dmGameObject::HInstance go = dmGameObject::New(m_Collection, "/dummy.goc");
 
     m_UpdateContext.m_DT = 0.25f;
-    dmhash_t id = hash("position");
-    dmGameObject::PropertyVar var(Vectormath::Aos::Vector3(10.f, 0.f, 0.f));
+    dmhash_t id1 = hash("position");
+    dmhash_t id2 = hash("scale");
+    dmhash_t id3 = hash("euler.z");
+    dmGameObject::PropertyVar var1(Vectormath::Aos::Vector3(10.f, 0.f, 0.f));
+    dmGameObject::PropertyVar var2(Vectormath::Aos::Vector3(2.0f));
+    dmGameObject::PropertyVar var3(360.0f);
     float duration = 1.0f;
     float delay = 0.f;
 
-    Animate(m_Collection, go, 0, id, dmGameObject::PLAYBACK_ONCE_FORWARD, var, dmEasing::Curve(dmEasing::TYPE_LINEAR), duration, delay, AnimationStopped, this, 0x0);
-    ASSERT_EQ(dmGameObject::PROPERTY_RESULT_OK, CancelAnimations(m_Collection, go, 0, id));
+    Animate(m_Collection, go, 0, id1, dmGameObject::PLAYBACK_ONCE_FORWARD, var1, dmEasing::Curve(dmEasing::TYPE_LINEAR), duration, delay, AnimationStopped, this, 0x0);
+    Animate(m_Collection, go, 0, id2, dmGameObject::PLAYBACK_ONCE_FORWARD, var2, dmEasing::Curve(dmEasing::TYPE_LINEAR), duration, delay, AnimationStopped, this, 0x0);
+    Animate(m_Collection, go, 0, id3, dmGameObject::PLAYBACK_ONCE_FORWARD, var3, dmEasing::Curve(dmEasing::TYPE_LINEAR), duration, delay, AnimationStopped, this, 0x0);
+    ASSERT_EQ(dmGameObject::PROPERTY_RESULT_OK, CancelAnimations(m_Collection, go, 0, id1));
+    ASSERT_EQ(dmGameObject::PROPERTY_RESULT_OK, CancelAnimations(m_Collection, go, 0, 0));
```

Comment:
I recommend adding a separate test for your new behavior.
The first reason is that this test is here and already working.

Secondly, you are not adding any tests in between these two CancelAnimation calls, which makes it possible you're not getting the result you wanted.

Reply:
Thanks for guidance, I'll make the changes. Based on what you wrote, it is a good idea to create a new e.g. cancel_all.script?